tests/Doctrine/Tests/DBAL/Functional/Platform/ColumnTest/IBMDB2.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Platform\ColumnTest;
+
+use Doctrine\DBAL\Platforms\DB2Platform;
+use Doctrine\Tests\DBAL\Functional\Platform\ColumnTest;
+
+final class IBMDB2 extends ColumnTest
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $this->requirePlatform(DB2Platform::class);
+    }
+
+    public function testVariableLengthStringNoLength() : void
+    {
+        self::markTestSkipped();
```

Comment:
Why are these skipped?

Reply:
Because DB2 does not allow `VARCHAR` without the length. `DB2Platform::testGetVariableLengthStringTypeDeclarationSQLNoLength()` proves proves that an attempt to create such a column will result in a `ColumnLengthRequired` exception, and we can run a functional test against it.

Reply:
The overall idea is that DBAL no longer provides the default lengths but it only forces you to specify the length if the target platform requires it. Therefore, if you only write against PostgreSQL and SQLite, you can skip the lengths. But if the same schema is deployed to MySQL, it will force you to specify the lengths.